server/plugin/hooks.go

```diff
@@ -56,19 +56,32 @@ func (p *SharePostPlugin) MessageWillBePosted(c *plugin.Context, post *model.Pos
 			return post, appErr.Error()
 		}
 		oldPostCreateAt := time.Unix(oldPost.CreateAt/1000, 0)
+		ateAt := time.Unix(oldPost.CreateAt/1000,0)
+
+		AuthorName :=postUser.GetDisplayNameWithPrefix(model.SHOW_NICKNAME_FULLNAME,"@")
+		fmtstmnt := "%s/api/v4/users/%s/image"
+		AuthorIcon :=fmt.Sprintf(fmtstmnt, *siteURL, oldPost.UserId)
+		if postUser.IsBot {
+			botUser := model.BotFromUser(postUser)
+			AuthorName = botUser.DisplayName
+			AuthorIcon = fmt.Sprintf(fmtstmnt, *siteURL, botUser.UserId)
+		}
+
 		attachment := []*model.SlackAttachment{
 			{
-				Timestamp:  oldPost.CreateAt,
-				AuthorName: postUser.GetDisplayNameWithPrefix(model.SHOW_NICKNAME_FULLNAME, "@"),
+				Timestamp:  ateAt,
```

Comment:
This line may cause the following error.

```
2020-05-03T17:08:21.002+0900    error   mlog/log.go:175 RPC call MessageWillBePosted to plugin failed.  {"plugin_id": "business.silly.quote", "error": "reading body EOF"}
2020-05-03T17:08:21.002+0900    info    go-plugin/stream.go:15  2020/05/03 17:08:21 rpc: gob error encoding body: gob: type not registered for interface: time.Time     {"plugin_id": "business.silly.quote", "source": "plugin_stderr"}
2020-05-03T17:08:21.061+0900    debug   app/post_metadata.go:113        Failed to get embedded content for a post       {"post_id": "xp5x6ksom3bizg37dcg35iz1gw", "error": "Get \"http://192.168.11.11:8065/tttt/pl/sy4k3pgpbbdczpkmpg61p71r3y\": address forbidden, you may need to set AllowedUntrustedInternalConnections to allow an integration access to your internal network"}
```

`Timestamp` can accept value of any type, but it must be `string` or `int64`.
https://github.com/mattermost/mattermost-server/blob/master/model/slack_attachment.go#L29